tecken/tests/test_upload.py

```diff
@@ -542,314 +312,155 @@ def mock_api_call(self, operation_name, api_params):
     )
 
 
-def test_key_existing_caching(botomock):
-    user = FakeUser("peterbe@example.com")
-    bucket_info = get_bucket_info(user)
-
-    sizes_returned = []
-    lookups = []
-
-    def mock_api_call(self, operation_name, api_params):
-        lookups.append((operation_name, api_params))
-
-        if operation_name == "HeadObject" and api_params["Key"] == "filename":
-            size = 1234
-            if sizes_returned:
-                size = 6789
-            result = {"ContentLength": size}
-            sizes_returned.append(size)
-            return result
-
-        raise NotImplementedError
-
-    client = bucket_info.client
-    with botomock(mock_api_call):
-        size, metadata = key_existing(client, "mybucket", "filename")
-        assert size == 1234
-        assert metadata is None
-        assert len(lookups) == 1
-
-        size, metadata = key_existing(client, "mybucket", "filename")
-        assert size == 1234
-        assert metadata is None
-        assert len(lookups) == 1
+def test_key_existing_caching(s3_helper):
+    s3_helper.create_bucket("publicbucket")
+    s3_helper.upload_fileobj(
+        bucket_name="publicbucket",
+        key="somefile.txt",
+        data=b"abc123",
+    )
 
-        key_existing.invalidate(client, "mybucket", "filename")
-        size, metadata = key_existing(client, "mybucket", "filename")
-        assert size == 6789
-        assert metadata is None
-        assert len(lookups) == 2
+    client = s3_helper.conn
 
+    size, metadata = key_existing(client, "publicbucket", "somefile.txt")
+    assert size == 6
+    assert metadata == {}
```

Comment:
The asserted metadata values for this test were wrong because we were testing against the mirror dimension.